src/coap_prng.c

```diff
@@ -20,37 +20,40 @@ coap_prng_default(void *buf, size_t len) {
 #ifdef HAVE_GETRANDOM
   return getrandom(buf, len, 0);
 #else /* !HAVE_GETRANDOM */
-  unsigned char *dst = (unsigned char *)buf;
-  while (len--)
-    *dst++ = rand() & 0xFF;
-  return 1;
+  return coap_prng_impl(buf,len);
 #endif /* !HAVE_GETRANDOM */
 }
 
 static coap_rand_func_t rand_func = coap_prng_default;
 
-void
-coap_set_prng(coap_rand_func_t rng) {
-  rand_func = rng;
-}
-
-void
-coap_prng_init(unsigned long seed) {
-#ifdef HAVE_GETRANDOM
-  /* No seed to seed the random source if getrandom() is used,
-   * see dtls_prng(). */
-  (void)seed;
-#else /* !HAVE_GETRANDOM */
-  srand(seed);
-#endif /* !HAVE_GETRANDOM */
-}
-
-int
-coap_prng(void *buf, size_t len) {
-  if (!rand_func) {
-    return 0;
-  }
-
-  rand_func(buf, len);
-  return 1;
-}
+#if defined(WITH_CONTIKI)
+
+#elif defined(WITH_LWIP) && defined(LWIP_RAND)
```

Comment:
This is the place to add in the Windows specific implementation taken from the coap_prng.h file.

Reply:
I will move coap_prng_impl() to here